### ISSUE TITLE
Add more documentation

### ISSUE THREAD

**Issue by Ladicek** [smallrye/jandex#133]
The existing Jandex documentation consists of a README, which is even somewhat outdated, and a Javadoc. This needs to be improved quite a bit. I'd personally downplay the published Javadoc (it's enough you have access to it in an IDE) and instead focus on a user guide kind of thing. We could use the SmallRye documentation site (https://smallrye.io/docs/), as we plan to move to SmallRye anyway (#124).

**Comment by n1hility** [comment]
FYI the javadoc is published to github pages atm: http://docs.wildfly.org/jandex/

**Comment by Ladicek** [comment]
I created a simple AsciiDoc/Antora-based documentation site in #242. It's currently almost equivalent to what used to be in the README, but I closed this issue anyway, as documentation is now much easier to add (and general docs improvements are always a long-running kind of thing). We'll refer to javadoc.io for javadocs, as we publish `-javadoc.jar` starting with Jandex 3.0 (actually 3.0.0-RC2 is already in Central and https://javadoc.io/doc/io.smallrye/jandex/latest/index.html works :-) ).
